js/libs/builder.js

```diff
@@ -363,38 +349,26 @@ var app = new Vue({
 
         var operation;
 
-        if ($vm.settings.dataStore.indexOf('editDataSource') > -1 || $vm.settings.dataSourceId) {
+        if ($vm.settings.dataStore.indexOf('editDataSource') > -1 && $vm.settings.dataSourceId) {
           var newHook = {
-            widgetInstanceId: $vm.settings.id,
+            widgetInstanceId: widgetId,
             runOn: ['update'],
             type: 'email',
             payload: $vm.settings.emailTemplateEdit
           };
 
           operation = Fliplet.DataSources.getById($vm.settings.dataSourceId).then(function(dataSource) {
-            if (dataSource.hooks.length) {
-              // Update existing hook
-              var currentHook = _.find(dataSource.hooks, function(o) {
-                return o.widgetInstanceId == $vm.settings.id;
-              });
+            // Remove old hook
+            dataSource.hooks = _.reject(dataSource.hooks, { type: 'email', widgetInstanceId: widgetId });
```

Comment:
Line 362: This can't be like like this, because there might be a hook for the Add template, and will match the same `type` and `widgetId` and you don't want to delete it.
Use the `runOn` key instead.